app/views/users/new.html.haml

```diff
@@ -1,33 +1,36 @@
-.container
-  %h4 Signup
-
-.container
-  .col-md-6.col-md-offset-3
-  = form_for @user do |f|
-    - if @user.errors.any?
-      %h2 Form is invalid
-      %ul
-        - @user.errors.full_messages.each do |message|
-          %li= message
-    - else
-      .field
-        = f.label :name
-        %br/
-        = f.text_field :name
-      .field
-        = f.label :email
-        %br/
-        = f.text_field :email
-      .field
-        = f.label :password
-        %br/
-        = f.password_field :password
-      .field
-        = f.label :password_confirmation
-        %br/
-        = f.password_field :password_confirmation
-        %br/
-        .actions= f.submit "Sign Up"
+.row
+  .col-xs-12.col-sm-8.col-md-4.col-sm-offset-2.col-md-offset-4
+    .panel.panel-default
+      .panel-heading
+        %h1.panel-title
+          Please Sign Up
+      .panel-body
+        %form{:role => "form"}
+          .row
+
+          = form_for @user do |f| 
+            - if @user.errors.any?
+            .alert.alert_danger
```

Comment:
Here is one issue. You don't have the alert tabbed over to be nested under the if statement.